app/models/repo/task.rb

```diff
@@ -10,6 +10,8 @@ class Task < ApplicationRecord
     belongs_to :challenge
     belongs_to :dependent_task, class_name: 'Repo::Task', optional: true
 
+    has_many :task_submissions, dependent: :destroy_async
```

Comment:
We can add a check for this assoc to task_spec as well

Reply:
good catch 👍, added it here -[add associations test to task](https://github.com/DevChallengeIT/devchallenge-platform/pull/32/commits/174af6c46e51c304b165a1223cc000b55b5fe20f)